api/handlers/service_offering.go

```diff
@@ -125,6 +150,7 @@ func (h *ServiceOffering) UnauthenticatedRoutes() []routing.Route {
 func (h *ServiceOffering) AuthenticatedRoutes() []routing.Route {
 	return []routing.Route{
 		{Method: "GET", Pattern: ServiceOfferingPath, Handler: h.get},
+		{Method: "PATCH", Pattern: ServiceOfferingPath, Handler: h.update},
```

Comment:
As this introduces a new endpoint, we would like to have an basic e2e test to ensure that the endpoint generally works. 

[Here](https://github.com/cloudfoundry/korifi/blob/9090f8b1d77c3442e0e1f42df2774b831c69dec8/tests/e2e/processes_test.go#L177-L198) is a similar test for patching processes metadata